client-cli/src/main/java/au/org/consumerdatastandards/client/cli/ReferenceTest.java

```diff
@@ -60,4 +75,19 @@ public void validatePath(@ShellOption(value = "-f", help = "payload file or fold
             }
         }
     }
+
+    @ShellMethod("List top-level payload models")
+    public void listPayloadModels() {
+        Set<String> topLevelModels = new TreeSet<>();
+        ConformanceModel conformanceModel = payloadValidator.getConformanceModel();
+        for (Class<?> payloadModel : conformanceModel.getPayloadModels()) {
+            Payload.Type payloadType = conformanceModel.getPayload(payloadModel).getPayloadType();
+            if (Payload.Type.RESPONSE_BODY.equals(payloadType)) {
+                topLevelModels.add(payloadModel.getSimpleName());
+            }
+        }
+        for (String topLevelModel : topLevelModels) {
+            LOGGER.info(topLevelModel);
```

Comment:
Wouldn't it be better for `listPayloadModels()` to return String rather than void? This way you could just build the output (using something like `StringBuilder` or better `PrintWriter` wrapping `StringWriter`) and then just return the resulting String.

Reply:
`listPayloadModels` is a `@ShellMethod` which is called by Spring Shell framework when I type `list-payload-models` on spring shell prompt. I don't have a chance to read the return value, as the method is not invoked by me. Please correct me if I am wrong

Reply:
`@ShellMethod`s can return any type - it will be printed as the output of the command. See: https://docs.spring.io/spring-shell/docs/current/reference/htmlsingle/ I think it's better to use this mechanism for command output rather than LOGGER.info given that the logger output is not predetermined.

Reply:
awesome, how handy that is. I'll change it to that way

Reply:
The change has been made. `PrintWriter` saves me from appending a new line by `println`, which is why I choose it instead of `StringBuilder`.
https://stackoverflow.com/questions/2980805/string-assembly-by-stringbuilder-vs-stringwriter-and-printwriter

Reply:
Yep, that's why I like PrintWriter as well. I guess we can gradually replace `LOGGER.info` in other `@ShellMethod`s such as `listProducts()`.

Reply:
I just had a look at `listProducts` and `getProductDetail`, in which some error message is logged or some info is logged that is not `products` or `productDetail` themselves. If we mix them as return value of `ShellMethod` . It seems semantically not clear.

Reply:
In Spring Shell the error is taken from the thrown exception message. No need to log it as well. The only problem I see with `LOGGER.debug` and `LOGGER.warn` laying around. Those can be replaced with the conditional inclusion by a "verbose" option of the command, e.g. it can be "-v" for warn and "-vv" for warn + debug. Also ANSI codes can be used to display text in different colours, but that is an overkill imho.

Reply:
Good to know that error can be taken from exception message, but things like `LOGGER.info("User agent specified as {}", apiClientOptions.getUserAgent());` cannot be easily fit into return value or exception message.

Reply:
I will merge this PR for now and leave any further optimisation for future. Get the codegen templates correct has higher priority anyway.

Reply:
Yep, we can look into the LOGGER output as a part of the future work on client-cli, surely we can do it, that's what I meant by _gradually_ replacing that.